client/shipping-label/views/purchase/index.js

```diff
@@ -43,6 +53,22 @@ const PrintLabelDialog = ( props ) => {
 		return props.labelActions.purchaseLabel;
 	};
 
+	const buttons = [
+		{
+			isDisabled: ! props.form.needsPrintConfirmation && ( ! props.canPurchase || props.form.isSubmitting ),
+			onClick: getPurchaseButtonAction(),
+			isPrimary: true,
+			label: getPurchaseButtonLabel(),
+		},
+	];
+
+	if ( ! props.form.needsPrintConfirmation ) {
```

Comment:
It's OK if the modal gets closed - labels can still be (re)printed using the link in the metabox.